File: exercism/typescript/clock/clock.ts
export class Clock {
  _hour: number;
  _minute: number;

  constructor(hour: number, minute?: number) {
    let minutes: number = hour * 60 + (minute ?? 0);
    if (minutes < 0) {
      minutes = 1440 - Math.abs(minutes) % 1440;
    }
    this._hour = Math.floor(minutes / 60) % 24;
    this._minute = minutes % 60;
  }

  public toString(): string {
    let h: number = this._hour;
    let m: number = this._minute;
    return `${h < 10 ? "0" + h : h}:${m < 10 ? "0" + m : m}`;
  }

  public plus(minutes: number): Clock {
    return new Clock(this._hour, this._minute + minutes);
  }

  public minus(minutes: number): Clock {
    return new Clock(this._hour, this._minute - minutes);
  }

  public equals(other: Clock): boolean {
    return this._hour === other._hour && this._minute === other._minute;
  }
}

File: exercism/python/clock/clock.py
class Clock:
    def __init__(self, hour, minute):
        self.minutes = minute % 60
        self.hour = (hour + minute // 60) % 24

    def __repr__(self) -> str:
        return f"Clock({self.hour}, {self.minutes})"

    def __str__(self) -> str:
        hour: str = f"0{self.hour}" if self.hour < 10 else self.hour
        minute: str = f"0{self.minutes}" if self.minutes < 10 else self.minutes
        return f"{hour}:{minute}"

    def __eq__(self, other) -> bool:
        return self.hour == other.hour and self.minutes == other.minutes

    def __add__(self, minutes):
        return Clock(self.hour, self.minutes + minutes)

    def __sub__(self, minutes):
        return Clock(self.hour, self.minutes - minutes)
